src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -400,6 +408,9 @@ private static function remove_json_comments( $array ) {
 	 * @return array Custom Post Type for the user's origin config.
 	 */
 	public static function get_user_data_from_wp_global_styles( $theme, $create_post = false, $post_status_filter = array( 'publish' ) ) {
+		if ( ! static::theme_has_support() ) {
+			return array();
+		}
```

Comment:
Good pointer, although the problem is in fact slightly different here. The order doesn't matter as much as the fact that `theme_has_support()` only supports the currently active theme, however the `$theme` parameter could be any theme (although in all use-cases in production code and even the tests, it _is_ the currently active theme).

I've therefore updated the logic here to only short-circuit if `$theme` is in fact the active theme. Preferably we should eventually add support for any theme to the `theme_has_support()` method to make this more stable, although one could also argue that's kind of pointless since I can't see a situation where you would want to call this for a non-active theme.